### PR TITLE
Make startup config optional

### PR DESCRIPTION
**Pull request by maturar**
Issues: #34

# Definition of Done

The PR shall be merged only if all items mentioned in [CONTRIBUTING.md](https://github.com/eclipse-ankaios/ankaios/blob/main/CONTRIBUTING.md#how-to-contribute) have been followed. In case an item is not applicable as described, please provide a short explanation in the description.

- [x] All steps in [CONTRIBUTING.md](https://github.com/eclipse-ankaios/ankaios/blob/main/CONTRIBUTING.md#how-to-contribute) have been handled

### CODE REVIEW

File: doc/docs/usage/quickstart.md [hunk 1]
```diff
@@ -4,7 +4,9 @@ If you have not installed Ankaios, please follow the instructions
 [here](installation.md). The following examples assumes that the
 installation script has been used with default options.
 
-Ankaios needs a startup configuration that contains all the workloads and their
+You can start workloads in Ankaios in several ways.
+For example you may define a file with the startup configuration and start Ankaios with systemd.
+The startup configuration contains all the workloads and their
 configuration which should be started when Ankaios starts up.
```
Comment (windsource) [inline doc/docs/usage/quickstart.md:10]:
```suggestion
You can start workloads in Ankaios in a number of ways.
For example, you can define a file with the startup configuration and use systemd to start Ankaios.
The startup configuration file contains all of the workloads and their configuration that you want to be started when Ankaios is started.
```

File: server/doc/swdesign/README.md [hunk 2]
```diff
@@ -88,6 +89,21 @@ Tags:
 Needs:
 - impl
 
+### Server starts without startup config
+`swdd~server-starts-without-startup-config~1`
+
+The Ankaios Server shall be able to start without startup configuration file.
```
Comment (krucod3) [inline server/doc/swdesign/README.md:95]:
```suggestion
When the Ankaios server is started without a startup config, the server shall start with an empty current state.
```

File: server/doc/swdesign/plantuml/seq_startup.plantuml [hunk 3]
```diff
@@ -42,6 +42,7 @@ tonic_server -> tonic_server ++--: start listening for state change requests
 grpc_server -> grpc_server ++--: start listening for execution requests /'evil laughter'/
 
 state_loader -> state_loader ++: load startup config
+note right: the startup config\nis provided either\nvia config file or the CLI
```
Comment (krucod3) [inline server/doc/swdesign/plantuml/seq_startup.plantuml:45]: The startup config cannot be provided by the CLI. If there was no startup config, a empty state is loaded which can subsequently be modified.

File: server/src/main.rs [hunk 4]
```diff
@@ -72,18 +82,23 @@ async fn main() -> Result<(), BoxedStdError> {
...
-                    request_id: "".to_owned(),
-                    startup_state: State::default(),
-                    current_state: state,
-                    workload_states: vec![],
-                },
-                vec![],
-            )
-            .await
-            .unwrap_or_illegal_state();
+        if let Some(state) = state {
```
Comment (krucod3) [inline server/src/main.rs:85]: We can omit the spawn here by checking beforehand if the state is some.

File: tests/stests/workloads/update_workload_podman.robot [hunk 5]
```diff
@@ -1,3 +1,4 @@
+*** Comments ***
```
Comment (krucod3) [inline tests/stests/workloads/update_workload_podman.robot:1]: Well .... this are not really comments ...

File: tests/stests/workloads/update_workload_podman.robot [hunk 6]
```diff
@@ -11,33 +12,55 @@
...
+    When user triggers "ank get state > ${new_state_yaml_file}"
     And user updates the state "${new_state_yaml_file}" with "currentState.workloads.nginx.runtimeConfig.commandOptions=['-p', '8082:80']"
     And user triggers "ank set state -f ${new_state_yaml_file} currentState.workloads.nginx"
     # Asserts
     Then the workload "nginx" shall have the execution state "Running" on agent "agent_A" within "30" seconds
     And the command "curl localhost:8082" shall finish with exit code "0" within "10" seconds
     [Teardown]    Clean up Ankaios
+
+# [stest->swdd~server-starts-without-startup-config~1]
+
```
Comment (krucod3) [inline tests/stests/workloads/update_workload_podman.robot:52]:
```suggestion
```

**Review by krucod3 — CHANGES_REQUESTED** [review]
Reviewed, findings documented.

File: server/src/main.rs (hunk 4 above)
Comment (maturar) [reply server/src/main.rs:85]: Yes, we can, but this implementation keeps the next line simple.
```
try_join!(communications_task, server_task, initial_state_task).unwrap_or_illegal_state();
```
This way the `try_join` has always three handles. Another approach would be to say that the `initial_state_task` is somehow optional. I am searching for possible solution.

Comment (krucod3) [reply server/src/main.rs:85]: Yes, the initial state task is only a workaround for now. We can even omit waiting for it after it is spawned.

Comment (maturar) [reply server/src/main.rs]: fixed

**Comment by maturar** [comment]
All findings have been fixed. The review can continue.

**Review by krucod3 — APPROVED** [review]
LGTM 👍

**Pull request merged**